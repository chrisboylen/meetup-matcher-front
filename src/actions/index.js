export const displayMeetups = (meetups) => ({
  type: 'DISPLAY_MEETUPS',
  meetups
});

export const loginUser = (user) => ({
  type: 'LOGIN_USER',
  user
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const userError = (message) => ({
  type: 'USER_ERROR',
  message
});