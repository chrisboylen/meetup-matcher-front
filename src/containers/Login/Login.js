import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { NavLink, withRouter } from 'react-router-dom';
import { signInFirebase } from '../../firebase/auth';
import { getUserInfoFirebase } from '../../firebase/firebase';
import { userError, loginUser } from '../../actions';
import { auth } from '../../firebase/firebase';

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;

    this.setState({ [name]: value })
  }

  handleLoginSubmit = async (e) => {
    e.preventDefault();
    const { history, login } = this.props;
    const { email, password } = this.state;
    await signInFirebase(email, password);
    const userInfo = await getUserInfoFirebase(auth.currentUser.uid)

    if (!userInfo) {
      return userError('Email and/or Password do not match.');
    }

    login(userInfo)
    history.push('/user')
  }

  render() {
    const { email, password } = this.state;

    return(
      <div className="login-cont">
        <form 
          className="login-form"
          onSubmit={this.handleLoginSubmit}
        >
          <input 
            className="email-input"
            required
            type="email"
            name="email"
            placeholder="Email"
            value={email}
            onChange={this.handleChange}
          />
          <input 
            className="password-input"
            required
            type="password"
            name="password"
            placeholder="Password"
            value={password}
            onChange={this.handleChange}
          />
          <button>Submit</button>
        </form>
        <NavLink className="login-link" to='/signup'>
            Sign Up
        </NavLink>
      </div>
    )
  }
};

Login.propTypes = {
  login: PropTypes.func,
  history: PropTypes.object,
  userError: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
  login: (user) => dispatch(loginUser(user)),
  userError: (message) => dispatch(userError(message))
});

export default withRouter(connect(null, mapDispatchToProps)(Login));