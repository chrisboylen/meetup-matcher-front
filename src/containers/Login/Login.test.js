import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';
import { login, userError } from '../../actions/index';

describe('Login Component', () => {
  let wrapper;
  let mockEvent

  beforeEach(() => {
    wrapper = shallow(<Login />);
    mockEvent = {target: {value: 'yes'}};
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
