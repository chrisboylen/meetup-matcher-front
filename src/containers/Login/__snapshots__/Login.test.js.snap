// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match snapshot 1`] = `
<div
  className="login-cont"
>
  <withRouter(Connect(Header)) />
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <h1>
      Log In
    </h1>
    <input
      className="email-input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      className="password-input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <button>
      Login
    </button>
  </form>
</div>
`;
